src/main/connection_listener/mod.rs

```diff
@@ -147,17 +166,27 @@ impl<UID: Uid> ConnectionListener<UID> {
         loop {
             match self.listener.accept() {
                 Ok((socket, _)) => {
+                    let mut socket = TcpSock::wrap(socket);
+                    if let Err(e) = socket.set_decrypt_ctx(DecryptContext::anonymous_decrypt(
+                        self.our_pk,
+                        self.our_sk.clone(),
+                    )) {
+                        error!("Failed to set decryption context: {}", e);
```

Comment:
unless you are terminating the app, it should be `warn!` for logic-errors, unexpected-results-which-cannot-be-easily-explained-and-might-cause-UB-now (as is the case here) etc.

Reply:
mmk, where are have you read those definitions from though? :)

Reply:
Ah it's the general convention we are following really.

Reply:
Actually i did a quick search and [landed on this](https://stackoverflow.com/a/2031209/1060004). It sort of matches my take on the log levels, if you shift by 1, meaning that since the link has an extra level `FATAL` which we don't if you consider our-error=their-fatal, our-warn=their-error and so on (shift by 1) it seems pretty close to the suggestion here.